yellowbrick/text/tsne.py

```diff
@@ -130,43 +132,51 @@ class TSNEVisualizer(TextVisualizer):
     ax : matplotlib axes
         The axes to plot the figure on.
 
-    decompose : string or None
+    decompose : string or None, default: ``'svd'``
         A preliminary decomposition is often used prior to TSNE to make the
-        projection faster. Specify `"svd"` for sparse data or `"pca"` for
-        dense data. If decompose is None, the original data set will be used.
+        projection faster. Specify ``"svd"`` for sparse data or ``"pca"`` for
+        dense data. If None, the original data set will be used.
 
-    decompose_by : int
+    decompose_by : int, default: 50
         Specify the number of components for preliminary decomposition, by
         default this is 50; the more components, the slower TSNE will be.
 
-    classes : list of strings
+    labels : list of strings
         The names of the classes in the target, used to create a legend.
+        Labels must match names of classes in sorted order.
 
     colors : list or tuple of colors
         Specify the colors for each individual class
 
     colormap : string or matplotlib cmap
         Sequential colormap for continuous target
 
+    random_state : int, RandomState instance or None, optional, default: None
+        If int, random_state is the seed used by the random number generator;
+        If RandomState instance, random_state is the random number generator;
+        If None, the random number generator is the RandomState instance used
+        by np.random. The random state is applied to the preliminary
+        decomposition as well as tSNE.
+
     kwargs : dict
         Pass any additional keyword arguments to the TSNE transformer.
     """
 
-    def __init__(self, ax=None, decompose='svd', decompose_by=50, classes=None,
-               colors=None, colormap=None, **kwargs):
+    # NOTE: cannot be np.nan
+    NULL_CLASS = None
+
+    def __init__(self, ax=None, decompose='svd', decompose_by=50, labels=None,
+               classes=None, colors=None, colormap=None, random_state=None, **kwargs):
         """
         Initialize the TSNE visualizer with visual hyperparameters.
         """
         super(TSNEVisualizer, self).__init__(ax=ax, **kwargs)
 
-        # Visualizer parameters
-        self.classes_ = classes
-        self.n_instances_ = 0
-
         # Visual Parameters
-        # TODO: Only colors currently works to select the colors of classes.
```

Comment:
`resolve_colors` is now fixed, so both colormap and colors works as expected; though this will probably change in the future.